sor/src/main/java/com/bazaarvoice/emodb/sor/db/astyanax/CqlDataWriterDAO.java

```diff
@@ -203,33 +203,40 @@ private void deleteCompactedDeltas(ByteBuffer rowKey, WriteConsistency consisten
         ConsistencyLevel consistencyLevel = SorConsistencies.toCql(consistency);
 
         // the old statement should be removed in the next version
-        BatchStatement oldTableStatement = _writeToOld ? new BatchStatement(BatchStatement.Type.UNLOGGED) : null;
-        BatchStatement newTableStatement = _writeToNew ? new BatchStatement(BatchStatement.Type.UNLOGGED) : null;
-        ResultSetFuture oldTableFuture = null;
+        ResultSetFuture legacyTableFuture = null;
+        ResultSetFuture auditBatchFuture = null;
 
         if (historyList != null && !historyList.isEmpty()) {
-            AuditBatchPersister auditBatchPersister = CqlAuditBatchPersister.build(_writeToOld ? oldTableStatement : newTableStatement,
+            BatchStatement auditBatchStatement = new BatchStatement();
+            AuditBatchPersister auditBatchPersister = CqlAuditBatchPersister.build(auditBatchStatement,
                     placement.getDeltaHistoryTableDDL(), _changeEncoder, _auditStore);
             _auditStore.putDeltaAudits(rowKey, historyList, auditBatchPersister);
+            auditBatchFuture = session.executeAsync(auditBatchStatement);
         }
 
         // delete the old deltas & compaction records
-        if (_writeToOld) {
+        if (_writeToLegacyDeltaTable) {
+            BatchStatement legacyTableStatement = new BatchStatement(BatchStatement.Type.UNLOGGED);
             for (UUID change : changesToDelete) {
-                oldTableStatement.add(deleteStatement(placement.getDeltaTableDDL(), rowKey, change, consistencyLevel));
+                legacyTableStatement.add(deleteStatement(placement.getDeltaTableDDL(), rowKey, change, consistencyLevel));
             }
-            oldTableFuture = session.executeAsync(oldTableStatement);
+            legacyTableFuture = session.executeAsync(legacyTableStatement);
         }
 
-        if (_writeToNew) {
+        if (_writeToBlockedDeltaTable) {
+            BatchStatement newTableStatement = new BatchStatement(BatchStatement.Type.UNLOGGED);
             for (UUID change : changesToDelete) {
                 newTableStatement.add(deleteStatement(placement.getBlockedDeltaTableDDL(), rowKey, change, consistencyLevel));
             }
             session.executeAsync(newTableStatement).getUninterruptibly();
```

Comment:
Need to capture the `ResultSetFuture` for this async call and check it later.  As it is now this could fail with no validation.

Reply:
we do capture it via the `getUninterruptibly()` call at the end of the statement

Reply:
Ah, ok.  That said, if we're going to call `executeAsync(newTableStatement).getUninterruptibly()` then why not just call `.execute(newTableStatement)`

Reply:
good point. I'll make the change now